tools/snp.sh

```diff
@@ -515,19 +521,32 @@ get_guest_kernel_version() {
 }
 
 save_binary_paths() {
+  
   local guest_kernel_version=$(get_guest_kernel_version)
   GENERATED_INITRD_BIN="${SETUP_WORKING_DIR}/initrd.img-${guest_kernel_version}"
+  echo " Latest bug  ${guest_kernel_version}"
+  
 
   # Guest kernel file path points to standard kernel file path across different linux distribution
   local guest_kernel=$(echo $(realpath "${SETUP_WORKING_DIR}/AMDSEV/linux/guest/vmlinuz-${guest_kernel_version}"))
 
 # Save binary paths in source file
-cat > "${SETUP_WORKING_DIR}/source-bins" <<EOF
+if [ "$SVSM" = true ]; then
+    cat > "${SETUP_WORKING_DIR}/source-bins" <<EOF
 QEMU_BIN="${SETUP_WORKING_DIR}/AMDSEV/qemu/build/qemu-system-x86_64"
 OVMF_BIN="${SETUP_WORKING_DIR}/AMDSEV/ovmf/Build/AmdSev/DEBUG_GCC5/FV/OVMF.fd"
 INITRD_BIN="${GENERATED_INITRD_BIN}"
 KERNEL_BIN="${guest_kernel}"
 EOF
+else
+    cat > "${SETUP_WORKING_DIR}/source-bins" <<EOF
+QEMU_BIN="${SETUP_WORKING_DIR}/AMDSEV/qemu/build/qemu-system-x86_64"
+OVMF_BIN="${SETUP_WORKING_DIR}/AMDSEV/ovmf/Build/OvmfX64/DEBUG_GCC5/FV/OVMF.fd"
+INITRD_BIN="${GENERATED_INITRD_BIN}"
+KERNEL_BIN="${guest_kernel}"
+IGVM_FILE="${SETUP_WORKING_DIR}/AMDSEV/svsm/bin/coconut-qemu.igvm"
+EOF
+fi
```

Comment:
Where the two logical statements are doing almost identical things for the first 4/5 of the condition, it make sense to remove most of it from the conditional statement and only append the IGVM_FILE code to the end of it when it is present. Keep in mind I have not tested the following code, so it may need some massaging.

```suggestion
cat > "${SETUP_WORKING_DIR}/source-bins" <<EOF
QEMU_BIN="${SETUP_WORKING_DIR}/AMDSEV/qemu/build/qemu-system-x86_64"
OVMF_BIN="${SETUP_WORKING_DIR}/AMDSEV/ovmf/Build/AmdSev/DEBUG_GCC5/FV/OVMF.fd"
INITRD_BIN="${GENERATED_INITRD_BIN}"
KERNEL_BIN="${guest_kernel}"
EOF
if [ -z "$SVSM" ]
then
    cat >> "${SETUP_WORKING_DIR}/source-bins" <<EOF
IGVM_FILE="${SETUP_WORKING_DIR}/AMDSEV/svsm/bin/coconut-qemu.igvm"
EOF
fi
```

Reply:
Our main goal in adding SVSM is to avoid breaking the existing codebase. You can see differences in the OVMF path AMDSEV/ovmf/Build/OvmfX64/DEBUG_GCC5/FV/OVMF.fd, as well as with IGVM. Once Coconut starts using the upstream EDK2, this path will be overwritten. Your suggested changes are more extensive than what we currently have. Does it make sense to proceed with these changes?

Reply:
My point is, if you have only two changes, you only need to adjust those two things. You don't need to have all the rest of the stuff in your conditional statement. Yes, you need to fix this